internal/filter/filter.go

```diff
@@ -100,6 +127,21 @@ func (f *Filter) ModuleReconcilerNodePredicate(kernelLabel string) predicate.Pre
 	)
 }
 
+func ModuleNMCReconcilerNodePredicate() predicate.Predicate {
+	return predicate.And(
+		skipDeletions,
+		predicate.Or(nodeBecomesSchedulable, predicate.LabelChangedPredicate{}),
+	)
+}
+
+func ModuleNMCReconcileJobPredicate() predicate.Predicate {
+	return predicate.And(
+		skipDeletions,
+		skipCreations,
+		moduleJobSuccess,
+	)
```

Comment:
Why do we need to process failed job? We are only interested in kicking in the reconciliation in case there is a chance that an image was built/created

Reply:
Don't we want to log failed builds, at least?

Reply:
That is not related to this controller, this one is responsible for module-nmc reconciliation. Controller responsible for building/signing will log what is needed